firebase-ai/src/main/kotlin/com/google/firebase/ai/FirebaseAI.kt

```diff
@@ -19,33 +19,39 @@ package com.google.firebase.ai
 import android.util.Log
 import com.google.firebase.Firebase
 import com.google.firebase.FirebaseApp
-import com.google.firebase.annotations.concurrent.Blocking
-import com.google.firebase.app
-import com.google.firebase.appcheck.interop.InteropAppCheckTokenProvider
-import com.google.firebase.auth.internal.InternalAuthProvider
-import com.google.firebase.inject.Provider
 import com.google.firebase.ai.type.Content
 import com.google.firebase.ai.type.GenerationConfig
+import com.google.firebase.ai.type.GenerativeBackend
+import com.google.firebase.ai.type.GenerativeBackendEnum
 import com.google.firebase.ai.type.ImagenGenerationConfig
 import com.google.firebase.ai.type.ImagenSafetySettings
 import com.google.firebase.ai.type.InvalidLocationException
 import com.google.firebase.ai.type.LiveGenerationConfig
 import com.google.firebase.ai.type.PublicPreviewAPI
 import com.google.firebase.ai.type.RequestOptions
 import com.google.firebase.ai.type.SafetySetting
+import com.google.firebase.ai.type.ServiceDisabledException
 import com.google.firebase.ai.type.Tool
 import com.google.firebase.ai.type.ToolConfig
+import com.google.firebase.annotations.concurrent.Blocking
+import com.google.firebase.app
+import com.google.firebase.appcheck.interop.InteropAppCheckTokenProvider
+import com.google.firebase.auth.internal.InternalAuthProvider
+import com.google.firebase.inject.Provider
 import kotlin.coroutines.CoroutineContext
 
-/** Entry point for all _Vertex AI for Firebase_ functionality. */
-public class FirebaseVertexAI
+/** Entry point for all _Firebase AI_ functionality. */
+public class FirebaseAI
 internal constructor(
   private val firebaseApp: FirebaseApp,
+  private val backend: GenerativeBackend,
   @Blocking private val blockingDispatcher: CoroutineContext,
-  private val location: String,
   private val appCheckProvider: Provider<InteropAppCheckTokenProvider>,
   private val internalAuthProvider: Provider<InternalAuthProvider>,
 ) {
+  init {
+    validateLocation()
```

Comment:
Actually do the validation when trying to create a `GenerativeBackend` instance with the wrong location name.